components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeService.java

```diff
@@ -294,4 +261,94 @@ public Scope updateScope(Scope updatedScope) throws IdentityOAuth2ScopeException
         }
         return scopes;
     }
+
+    /**
+     * Scope validation before adding the scope.
+     *
+     * @param scope Scope.
+     * @throws IdentityOAuth2ScopeClientException
+     */
+    private void addScopePreValidation(Scope scope) throws IdentityOAuth2ScopeClientException {
+
+        validateScopeName(scope.getName());
+        validateDisplayName(scope.getDisplayName());
+    }
+
+    /**
+     * Do the validation before updating the scope.
+     *
+     * @param updatedScope Updated scope.
+     * @throws IdentityOAuth2ScopeClientException
+     */
+    private void updateScopePreValidation(Scope updatedScope) throws IdentityOAuth2ScopeClientException {
+
+        validateScopeName(updatedScope.getName());
+        validateDisplayName(updatedScope.getDisplayName());
+    }
+
+    /**
+     * Check whether scope name is provided or not.
+     *
+     * @param scopeName Scope name.
+     * @throws IdentityOAuth2ScopeClientException
+     */
+    private void validateScopeName(String scopeName) throws IdentityOAuth2ScopeClientException {
+
+        // Check whether the scope name is provided.
+        if (StringUtils.isBlank(scopeName)) {
+            throw Oauth2ScopeUtils.generateClientException(Oauth2ScopeConstants.ErrorMessages.
+                    ERROR_CODE_BAD_REQUEST_SCOPE_NAME_NOT_SPECIFIED, null);
+        }
+        validateWhiteSpaces(scopeName);
+    }
+
+    /**
+     * Check whether scope name contains any white spaces.
+     *
+     * @param scopeName Scope name.
+     * @throws IdentityOAuth2ScopeClientException
+     */
+    private void validateWhiteSpaces(String scopeName) throws IdentityOAuth2ScopeClientException {
+
+        // Check whether the scope name contains any white spaces.
+        Pattern pattern = Pattern.compile("\\s");
+        Matcher matcher = pattern.matcher(scopeName);
+        boolean foundWhiteSpace = matcher.find();
+
+        if (foundWhiteSpace) {
+            throw Oauth2ScopeUtils.generateClientException(Oauth2ScopeConstants.ErrorMessages.
+                    ERROR_CODE_BAD_REQUEST_SCOPE_NAME_CONTAINS_WHITESPACES, null);
```

Comment:
Can we pass the scope name here instead of null?

Reply:
+1

Reply:
fixed